src/client.js

```diff
@@ -78,8 +76,14 @@ function revoke_token (token, client_id, type = '') {
         .max_retry(config.retry_count)
         .promise()
         .then(result => {
-            if (key) {
+            const client_key = cache.find_key('client', {token});
+            const server_key = cache.find_key('server', {token});
+
+            if (client_key) {
                 cache.forget('client', client_key);
+            }
+
+            if (server_key) {
```

Comment:
Do we know if client_key and server_key are actually the same? If so, we can just manage one key?

Also, please add test cases for caching usage (which would catch the error I have identified)

Reply:
Hmmm. I think there are possibilities that the `client_key` and `server_key` is the same but then since accounts are used by other services, there's a huge chance that it would not be the same?

And sure. I'll add unit testing for caching usage.

Reply:
This is already updated. Kindly take time to review it .. Thank you. :smile: